diff --git a/libavformat/dashdec.c b/libavformat/dashdec.c
index 6a147c5e..6a147c5f 100644
--- a/libavformat/dashdec.c
+++ b/libavformat/dashdec.c
@@ -407,6 +407,9 @@ static int open_url(AVFormatContext *s, AVIOContext **pb, const char *url,
     int proto_name_len;
     int ret;
 
+    if (ff_check_interrupt(c->interrupt_callback))
+        return AVERROR_EXIT;
+
     if (av_strstart(url, "crypto", NULL)) {
         if (url[6] == '+' || url[6] == ':')
             proto_name = avio_find_protocol_name(url + 7);
diff --git a/libavformat/hls.c b/libavformat/hls.c
index 9c68dd8a..9c68dd8b 100644
--- a/libavformat/hls.c
+++ b/libavformat/hls.c
@@ -658,6 +658,9 @@ static int open_url(AVFormatContext *s, AVIOContext **pb, const char *url,
     int ret;
     int is_http = 0;
 
+    if (ff_check_interrupt(c->interrupt_callback))
+        return AVERROR_EXIT;
+
     if (av_strstart(url, "crypto", NULL)) {
         if (url[6] == '+' || url[6] == ':')
             proto_name = avio_find_protocol_name(url + 7);
